src/ucioption.cpp

```diff
@@ -159,7 +170,12 @@ Option& Option::operator=(const std::string& v) {
         currentValue = v;
 
     if (on_change)
-        on_change(*this);
+    {
+        const auto ret = on_change(*this);
+
+        if (ret && parent != nullptr && parent->info != nullptr)
+            parent->info(ret);
```

Comment:
law of demeter but not a big deal, personally I'd propagate this up with function calls to OptionsMap